src/rewriters/ArithmeticEqualityRewriter.h

```diff
@@ -8,10 +8,15 @@
 #include "Rewriter.h"
 
 class EqualityRewriterConfig : DefaultRewriterConfig {
+    std::unique_ptr<Map<PTRef,bool,PTRefHash>> notOkToPartition;
     LALogic & logic;
-
 public:
-    EqualityRewriterConfig(LALogic & logic): logic(logic) {}
+    EqualityRewriterConfig(LALogic & logic): logic(logic) { notOkToPartition = std::unique_ptr<Map<PTRef,bool,PTRefHash>>(new Map<PTRef,bool,PTRefHash>()); }
+
+    std::unique_ptr<Map<PTRef,bool,PTRefHash>> getAndClearNotOkToPartition() {
+        auto tmp = std::unique_ptr<Map<PTRef,bool,PTRefHash>>(new Map<PTRef,bool,PTRefHash>());
+        std::swap(tmp, notOkToPartition); return tmp;
+    }
```

Comment:
I am wondering if we need to clear it like that.
From the current use case, I would say no.
(I am just trying to avoid the allocation if it is not necessary.)

Reply:
I think it's safer this way.  The rewriter resets its state after this call to the initial state, and I think it's confusing otherwise (i.e., resetting to a state where `notOkToPartition` is `nullptr`).

Reply:
OK, I agree.